### ISSUE TITLE
lsp with a next checker doesn't show errors on first opening a file

### ISSUE THREAD

**Issue by RohanHart** [flycheck/flycheck#1891]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
With flycheck configured like this `(flycheck-add-next-checker 'lsp '(warning . javascript-eslint) 'append)` any messages at any level from lsp are not displayed by flycheck - only those from the next checker. When reverting the buffer I can see the lsp errors appear briefly before being replaced by the next checker's errors. Forcing a recheck with `flycheck-buffer` does correctly show the lsp errors. Changing the next checker to always run doesn't change this erroneous behaviour.

## Steps to reproduce
1. `(flycheck-add-next-checker 'lsp '(warning . javascript-eslint) 'append)`
2. Open the flycheck errors buffer
3. Open a typescript file with code and lint issues
4. Only lint issues are displayed in the errors buffer

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
All errors should be displayed after opening or revering a file buffer.

## System configuration
```
Syntax checkers for buffer earn.component.ts in typescript-mode:

First checker to run:

  lsp (explicitly selected)
    - may enable:    yes
    - may run:       t
    - next checkers: javascript-eslint

Checkers that may run as part of the first checker's chain:

  javascript-eslint
    - may enable:  yes
    - executable:  Found at [redacted]/node_modules/.bin/eslint
    - config file: found

Checkers that are compatible with this mode, but will not run until properly configured:

  typescript-tslint (automatically disabled) reset
    - may enable:         no
    - executable:         Not found
    - configuration file: Not found

The following syntax checkers are not registered:
  - python-pyright
Try adding these syntax checkers to `flycheck-checkers'.

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20210321.852)
Emacs version:    27.2
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [x] Other shared configuration

## Additional notes
Using lsp config from https://github.com/bbatsov/prelude with extra customisation

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
